app/scripts/components/analysis/define/use-stac-collection-search.ts

```diff
@@ -86,13 +105,15 @@ export function useStacCollectionSearch({
 
 function getInTemporalAndSpatialExtent(collectionData, aoi, timeRange) {
   const matchingCollectionIds = collectionData.reduce((acc, col) => {
-    const id = col.id;
+    const { id, stacApiEndpoint } = col;
 
     // Is is a dataset defined in the app?
     // If not, skip other calculations.
-    const isAppDataset = allAvailableDatasetsLayers.some(
-      (l) => l.stacCol === id
-    );
+    const isAppDataset = allAvailableDatasetsLayers.some((l) => {
+      const stacApiEndpointUsed =
+        l.stacApiEndpoint ?? process.env.API_STAC_ENDPOINT;
+      return l.stacCol === id && stacApiEndpointUsed === stacApiEndpoint;
```

Comment:
I'm having difficulty understanding why you need to check the stac endpoint to know whether is an app dataset.
Is it because there may be datasets with the same stacCol but different endpoints? 🤔

Reply:
Yes, this is for the case with the same stacCol from different endpoints. stacCol can't be a unique identifier anymore.